backend-api/src/functions/services/clientRegistryService/clientRegistryService.ts

```diff
@@ -0,0 +1,203 @@
+import { createHash } from "crypto";
+import { Result, errorResult, successResult } from "../../utils/result";
+import {
+  GetParameterCommand,
+  GetParameterRequest,
+  SSMClient,
+} from "@aws-sdk/client-ssm";
+import { ssmClient } from "./ssmClient";
+
+let cache: CacheEntry | null = null;
+
+export class ClientRegistryService
+  implements
+    IGetRegisteredIssuerUsingClientSecrets,
+    IGetPartialRegisteredClientByClientId
+{
+  ssmClient: SSMClient;
+  cacheTTL: number;
+
+  constructor() {
+    this.ssmClient = new SSMClient(ssmClient);
```

Comment:
I think the `ssmClient` argument here is already an instance of a `SSMClient`
```suggestion
    this.ssmClient = ssmClient;
```
Also, should the ssmClient be passed through the constructor as a default parameter?

Reply:
Existing issue but updated here :) Thanks for raising